src/crypto/tests/CHIPCryptoPALTest.cpp

```diff
@@ -1795,8 +1795,232 @@ static void TestPubkey_x509Extraction(nlTestSuite * inSuite, void * inContext)
+        0xab, 0x66, 0x62, 0x2e, 0x9d, 0x26, 0x10, 0x41, 0xf8, 0x0e, 0x97, 0x49, 0xfe, 0xff, 0x78, 0x10, 0x02, 0x49, 0x67, 0xae,
+        0xdf, 0x41, 0x38, 0x36, 0x5b, 0x0a, 0x22, 0x57, 0x14, 0x9c, 0x9a, 0x12, 0x3e, 0x0d, 0x30, 0xaa, 0x30, 0x02, 0x20, 0xe0,
+        0x42, 0x1b, 0x91, 0xc6, 0xfd, 0xcd, 0xb4, 0x0e, 0x2a, 0x4d, 0x2c, 0xf3, 0x1d, 0xb2, 0xb4, 0xe1, 0x8b, 0x41, 0x1b, 0x1d,
+        0x3a, 0xd4, 0xd1, 0x2a, 0x9d, 0x90, 0xaa, 0x8e, 0x52, 0xfa, 0xe2, 0x26, 0x03, 0xfd, 0xc6, 0x5b, 0x28, 0x30, 0x05, 0x17,
+        0x73, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x5f, 0x76, 0x65, 0x6e, 0x64, 0x6f, 0x72, 0x5f, 0x72, 0x65, 0x73, 0x65, 0x72, 0x76,
+        0x65, 0x64, 0x31, 0x30, 0x07, 0x18, 0x76, 0x65, 0x6e, 0x64, 0x6f, 0x72, 0x5f, 0x72, 0x65, 0x73, 0x65, 0x72, 0x76, 0x65,
+        0x64, 0x33, 0x5f, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x18
+    };
+
+    attestationElementsLen = sizeof(certificationDeclaration) + sizeof(attestationNonce) + sizeof(timestamp) +
+        sizeof(vendorReserved1) + sizeof(vendorReserved3) + sizeof(uint64_t) * 5;
+    attestationElements.Alloc(attestationElementsLen);
+    NL_TEST_ASSERT(inSuite, attestationElements);
+
+    {
+        MutableByteSpan attestationElementsSpan(attestationElements.Get(), attestationElementsLen);
+        err = ConstructAttestationElements(ByteSpan(certificationDeclaration), ByteSpan(attestationNonce), timestamp, ByteSpan(),
+                                           ByteSpan(vendorReserved1), ByteSpan(), ByteSpan(vendorReserved3), ByteSpan(),
+                                           attestationElementsSpan);
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+        attestationElementsLen = attestationElementsSpan.size();
+    }
+
+    NL_TEST_ASSERT(inSuite, sizeof(attestationElementsTestVector) == attestationElementsLen);
+    NL_TEST_ASSERT(inSuite,
+                   memcmp(attestationElements.Get(), attestationElementsTestVector, sizeof(attestationElementsTestVector)) == 0);
+
+    ByteSpan certificationDeclarationSpan;
+    ByteSpan attestationNonceSpan;
+    uint32_t timestampDeconstructed;
+    ByteSpan firmwareInfoSpan;
+    ByteSpan vendorReserved1Span;
+    ByteSpan vendorReserved2Span;
+    ByteSpan vendorReserved3Span;
+    ByteSpan vendorReserved4Span;
+    err = DeconstructAttestationElements(ByteSpan(attestationElementsTestVector), certificationDeclarationSpan,
+                                         attestationNonceSpan, timestampDeconstructed, firmwareInfoSpan, vendorReserved1Span,
+                                         vendorReserved2Span, vendorReserved3Span, vendorReserved4Span);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    NL_TEST_ASSERT(inSuite, sizeof(certificationDeclaration) == certificationDeclarationSpan.size());
+    NL_TEST_ASSERT(inSuite,
+                   memcmp(certificationDeclarationSpan.data(), certificationDeclaration, sizeof(certificationDeclaration)) == 0);
```

Comment:
Might be simpler as:
```suggestion
    NL_TEST_ASSERT(inSuite, ByteSpan(certificationDeclaration).data_equal(certificationDeclarationSpan));
```
but either way, I guess.